fix: Message formatting breaks telegram > discord #340 | feat: new option "useEmbeds"

Exzender opened the pull request:
1. Full fix for #340 (When some entity in Telegram has double(triple...) formatting - parser cuts off part of source message)
2. Fix: Message edits in telegram thread were not transferred
3. Added bridge option "useEmbeds" to define embed usage (always / never / auto)

Commit:
fix for message cutting on complex message formatting (#340)

Commit:
fix for message cutting on complex message formatting (#340)

Commit:
Fix: Editions from a telegram topic/thread not translated to the discord thread

Commit:
Added option 'useEmbeds' to Discord section in bridge config. Can be 'always', 'never', 'auto' (default = auto)

Exzender commented:
Added bridge option "useEmbeds" to disable/enable embed usage.

Lulalaby commented:
Well, amazing
![image](https://github.com/TediCross/TediCross/assets/14029133/c66908ff-78bc-4b11-b9cd-7b6f42eb630a)

Commit:
fixed: list, headings markdown parsing
added: link markdown parsing (#262)

Exzender commented:
> Well, amazing ![image](https://user-images.githubusercontent.com/14029133/261850316-c66908ff-78bc-4b11-b9cd-7b6f42eb630a.png)

fixed.
and added link parsing (from Discord) (#262)

Exzender committed:
Merge branch 'TediCross:dev' into dev

Commit:
Reworked telegram entities parser to support nested/overlapped tags (#340)

Exzender commented:
Сompletely rewrote the code of handleEntities to support nested entities (#340)

Lulalaby commented:
btw it seems your local git config is incorrect (either user or email), you might wanna fix it

Lulalaby commented on README.md:
```diff
@@ -102,6 +102,7 @@ As mentioned in the step-by-step installation guide, there is a settings file. H
 	* `discord.relayLeaveMessages`: Whether to relay messages to Telegram about people leaving the Discord chat
 	* `discord.sendUsernames`: Whether to send the sender's name with the messages to Telegram
 	* `discord.crossDeleteOnTelegram`: Whether to also delete the corresponding message on Telegram when one is deleted in Discord
+    * `discord.useEmbeds`: Whether to use embeds for current bridge. Can be `always`, `never`, `auto`. Defaults to `false`
 	* `threadMap`: An array containing all threads mapping for each bridge
```
```suggestion
	* `discord.crossDeleteOnTelegram`: Whether to also delete the corresponding message on Telegram when one is deleted in Discord
        * `discord.useEmbeds`: Whether to use embeds for current bridge. Can be `always`, `never`, `auto`. Defaults to `false`
	* `threadMap`: An array containing all threads mapping for each bridge
```

Lulalaby committed:
Update README.md

Signed-off-by: Lala Sabathil <aiko@aitsys.dev>

Lulalaby committed:
fix: make the bot great again :sweat:

Fixed that certain messages won't be relayed
Fixed eslint problems
Removed option for relayCommands (this was breaking stuff)

Lulalaby commented on src/telegram2discord/setup.ts:
```diff
@@ -150,6 +148,7 @@ export function setup(
 
 			// Don't crash on errors
 			tgBot.catch((err: any) => {
+				console.log(err);
```
```suggestion
```

Lulalaby commented on src/telegram2discord/setup.ts:
```diff
@@ -130,8 +130,6 @@ export function setup(
 			tgBot.use(middlewares.removeD2TBridges);
-			//@ts-ignore telegram expacts a second parameter
-			tgBot.command(middlewares.removeBridgesIgnoringCommands);
 			tgBot.on("new_chat_members", middlewares.removeBridgesIgnoringJoinMessages);
```
```suggestion
			//@ts-ignore telegram expacts a second parameter
			//tgBot.command(middlewares.removeBridgesIgnoringCommands);
			tgBot.on("new_chat_members", middlewares.removeBridgesIgnoringJoinMessages);
```

Lulalaby committed:
Apply suggestions from code review

Signed-off-by: Lala Sabathil <aiko@aitsys.dev>
